src/panels/config/integrations/integration-panels/zha/zha-network-visualization-page.ts

```diff
@@ -194,6 +236,59 @@ export class ZHANetworkVisualizationPage extends LitElement {
     return label;
   }
 
+  private _handleSearchChange(ev: CustomEvent) {
+    this._filter = ev.detail.value;
+    if (!this._network) {
+      return;
+    }
+    if (this._filter && this._filter !== "") {
+      const filteredNodeIds: (string | number)[] = this._nodes
+        .filter(
+          (node: Node) =>
+            node.label &&
+            node.label.toLowerCase().includes(this._filter!.toLowerCase())
+        )
+        .map((node: Node) => node.id!);
```

Comment:
And instead of a filter and map, use a `forEach` that will add the id to an array